pkg/controller/master/node/promote_controller.go

```diff
@@ -91,6 +97,20 @@ func PromoteRegister(ctx context.Context, management *config.Management, options
 	return nil
 }
 
+func fetchPromoteImage(appClient catalogv1.AppClient, namespace, name string) error {
```

Comment:
Is it better to return promote image name rather than storing in a variable which can be modified by code in `node` package.

Reply:
Yes, it makes more sense. I could also remove `fetchPromoteImage` function because I could just directly call utilCatalog to retrieve image name.